Shared/Core/ArtworkService.swift

```diff
@@ -10,76 +10,92 @@ import Foundation
 import UIKit
 
 final class ArtworkService {
-    private let cache: Cachable
-    private let session: WebSession
+    static var shared: ArtworkService = {
+        return ArtworkService(fetchers: [
+            CachedArtworkFetcher(cache: Cache.WXYC),
+            RemoteArtworkFetcher(locator: DiscogsLocator()),
+            RemoteArtworkFetcher(locator: LastFMLocator()),
+            RemoteArtworkFetcher(locator: iTunesLocator()),
+            DefaultArtworkFetcher()
+            ])
+    }()
     
-    init(cache: Cachable = Cache.WXYC, session: WebSession = URLSession.shared) {
-        self.cache = cache
-        self.session = session
+    private let fetchers: [ArtworkFetcher]
+    
+    private init(fetchers: [ArtworkFetcher]) {
+        self.fetchers = fetchers
     }
     
     func getArtwork(for playcut: Playcut) -> Future<UIImage> {
-        return self.getCachedArtwork()
-            || self.getNetworkArtwork(for: playcut)
-            || self.getDefaultArtwork()
+        let (first, rest) = (self.fetchers.first!, self.fetchers.dropFirst())
+        return rest.reduce(first.getArtwork(for: playcut), { $0 || $1.getArtwork(for: playcut) })
```

Comment:
I know this isn't the most readable code. Sorry. Let me know how I can improve this